DSCResources/MSFT_xDhcpServerExclusionRange/MSFT_xDhcpServerExclusionRange.psm1

```diff
@@ -0,0 +1,319 @@
+        [String]$ScopeId,
+
+        [parameter(Mandatory)]
+        [String]$StartRange,
+
+        [parameter(Mandatory)]
+        [String]$EndRange,
+
+        [ValidateSet('Present','Absent')]
+        [String]$Ensure = 'Present',
+
+        [Switch]$Apply
+    )
+
+    $checkExclusionRangeMessage = $LocalizedData.CheckExclusionRangeMessage
+    Write-Verbose -Message $checkExclusionRangeMessage
+
+    $dhcpExclusionRange = Get-DhcpServerv4ExclusionRange | ? {($_.ScopeId -eq $ScopeId) -and ($_.StartRange -eq $StartRange) -and ($_.EndRange -eq $EndRange)}
+
+    # Initialize the parameter collection
+    if($Apply)
+    {
+        $parameters = @{}
+    }
+
+    # dhcpExclusionRange is set
+    if($dhcpExclusionRange)
+    {
+        $TestExclusionRangeMessage = $($LocalizedData.TestExclusionRangeMessage) -f 'present', $Ensure
+        Write-Verbose -Message $TestExclusionRangeMessage
+
+        # if it should be present, test individual properties to match parameter values
+        if($Ensure -eq 'Present')
+        {
+            #region Test the ScopeId
+            $checkPropertyMsg = $($LocalizedData.CheckPropertyMessage) -f 'scopeid'
+            Write-Verbose -Message $checkPropertyMsg
+
+            if($dhcpExclusionRange.ScopeId -ne $ScopeId)
+            {
+                $notDesiredPropertyMsg = $($LocalizedData.NotDesiredPropertyMessage) -f 'scopeid',$ScopeId,$($dhcpExclusionRange.ScopeId)
+                Write-Verbose -Message $notDesiredPropertyMsg
+
+                if($Apply)
+                {
+                    $parameters['ScopeId'] = $ScopeId
+                }
+                else
+                {
+                    return $false
+                }
+            }
+            else
+            {
+                $desiredPropertyMsg = $($LocalizedData.DesiredPropertyMessage) -f 'scopeid'
+                Write-Verbose -Message $desiredPropertyMsg
+            }
+            #endregion scopeid
+
+            if($Apply)
+            {
+                # If parameters contains more than 0 key, set the DhcpServer scope
+                if($parameters.Count -gt 0)
+                {
+                    Add-DhcpServerv4ExclusionRange @parameters -ScopeId $dhcpExclusionRange.ScopeId
+                    Write-PropertyMessage -Parameters $parameters -keysToSkip ScopeId `
+                                          -Message $($LocalizedData.SetPropertyMessage) -Verbose
+                }
+            } # end Apply
+            else
+            {
+                return $true
+            }
+        } # end ensure eq present
+
+        # If dhcpExclusionRange should be absent
+        else
+        {
+            if($Apply)
+            {
+                $removingExclusionRangeMsg = $LocalizedData.RemovingExclusionRangeMessage
+                Write-Verbose -Message $removingExclusionRangeMsg
+
+                # Remove the scope
+                Remove-DhcpServerv4ExclusionRange -ScopeId $dhcpExclusionRange.ScopeId
```

Comment:
Does Remove-DhcpServerv4ExclusionRange prompt for confirmation (it should do!)? If so, we'll need a -confirm:$false?